pyam/index.py

```diff
@@ -63,8 +63,8 @@ def replace_index_values(df, level, mapping):
 def append_index_level(index, codes, level, name, order=False):
     """Append a level to a pd.MultiIndex"""
     new_index = pd.MultiIndex(
-        codes=index.codes + [codes],
-        levels=index.levels + [level],
+        codes=index.codes + [[codes] * len(index.codes[0])],
```

Comment:
If there exists tests for this function, would it be possible to add one that would catch this case moving forward (i.e., which would have failed without this new update)? Or can you confirm that `test_append_index()` does so? Thanks!

Reply:
The function wasn't actually operational and there were no tests before this PR. So now the function works and the test `test_append_index()` ensures that it works as expected going forward.